cmd/nerdctl/create_linux_test.go

```diff
@@ -33,3 +36,69 @@ func TestCreate(t *testing.T) {
 	base.Cmd("start", tID).AssertOK()
 	base.Cmd("logs", tID).AssertOutContains("foo")
 }
+
+func TestCreateWithMACAddress(t *testing.T) {
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+	networkBridge := "testNetworkBridge" + tID
+	networkMACvlan := "testNetworkMACvlan" + tID
+	networkIPvlan := "testNetworkIPvlan" + tID
+	base.Cmd("network", "create", networkBridge, "--driver", "bridge").AssertOK()
+	base.Cmd("network", "create", networkMACvlan, "--driver", "macvlan").AssertOK()
+	base.Cmd("network", "create", networkIPvlan, "--driver", "ipvlan").AssertOK()
+	t.Cleanup(func() {
+		base.Cmd("network", "rm", networkBridge).AssertOK()
+		base.Cmd("network", "rm", networkMACvlan).AssertOK()
+		base.Cmd("network", "rm", networkIPvlan).AssertOK()
+	})
+	tests := []struct {
+		Network string
+		Ok      bool
+	}{
+		{"host", false},
+		{"none", true},
+		{"container:whatever" + tID, false},
+		{"bridge", true},
+		{networkBridge, true},
+		{networkMACvlan, true},
+		{networkIPvlan, false},
+	}
+	for i, test := range tests {
+		macAddress, err := nettestutil.GenerateMACAddress()
+		if err != nil {
+			t.Errorf("failed to generate MAC address: %s", err)
+		}
+		containerName := fmt.Sprintf("%s_%d", tID, i)
```

Comment:
nits: you can use `testutil.Identifier(t)` to get an unique indentify

Reply:
tID = `testutil.Identifier(t)`